shell/common/engine.cc

```diff
@@ -521,24 +521,28 @@ void Engine::ScheduleSecondaryVsyncCallback(uintptr_t id,
 }
 
 void Engine::HandleAssetPlatformMessage(fml::RefPtr<PlatformMessage> message) {
-  fml::RefPtr<PlatformMessageResponse> response = message->response();
-  if (!response) {
-    return;
-  }
-  const auto& data = message->data();
-  std::string asset_name(reinterpret_cast<const char*>(data.data()),
-                         data.size());
-
-  if (asset_manager_) {
-    std::unique_ptr<fml::Mapping> asset_mapping =
-        asset_manager_->GetAsMapping(asset_name);
-    if (asset_mapping) {
-      response->Complete(std::move(asset_mapping));
-      return;
-    }
-  }
-
-  response->CompleteEmpty();
+  task_runners_.GetIOTaskRunner()->PostTask(fml::MakeCopyable(
+      [message = std::move(message), asset_manager = asset_manager_]() mutable {
+        fml::RefPtr<PlatformMessageResponse> response = message->response();
+        if (!response) {
+          return;
+        }
```

Comment:
Seems like this early out should happen on the UI thread since it should be low cost.  What line of this code is the heaviest?  GetAsMapping?  I'd only defer to the IO thread if we are going to hit that since posting to another thread incurs extra cost.